src/web/components/SharingPermission/BulkAddPermissions.tsx

```diff
@@ -12,28 +14,32 @@ import {
   getDefaultDataProviderCheckboxState,
   getDefaultDSPCheckboxState,
   getDefaultPublisherCheckboxState,
+  getFilteredParticipantsByType,
   getRecommendationMessageFromTypeNames,
   getRecommendedTypeFromParticipant,
   hasUncheckedASharedType,
 } from './bulkAddPermissionsHelpers';
+import { ParticipantItemSimple } from './ParticipantItem';
 
 import './BulkAddPermissions.scss';
 
 type BulkAddPermissionsProps = {
   participant: ParticipantDTO | null;
   hasSharingParticipants: boolean;
+  availableParticipants: AvailableParticipantDTO[];
```

Comment:
I guess these participants here are getting back from admin service, do they AvailableParticipantDTO type?

Reply:
good question - these participants need to be driven from the sites from admin service. my plan is for sharingPermissions.tsx (or elsewhere) to map these sites to AvailableParticipantDTOs (or similar), then pass them into this component. We need to do this because the source of truth for the SiteId comes from admin, but the display name for the participant lives in our db.  So admin doesn't need to have a notion of AvailableParticipantDTO. Does that sound reasonable?

Reply:
I understand there will eventually be 1 to 1 relationship between the site and the participant. But if a participant owner didn't create an account in self-serve portal, I think we probably still want to display that site in this list. Because technically when current participant share with this type, they will share with all sites no matter they have account in self-serve portal?

Reply:
that is correct. do you see any issue with this implementation to achieve that? I understand that it may not necessarily use `AvailableParticipantDTO` type but it would be something similar. Or maybe could have a different name for the prop for clarity?

Reply:
What I mean is if we don't have such site lives in our db, how are we going to map it to the AvailableParticipantDTOs? do you reformat the object that we get from admin service to AvailableParticipantDTO type

Reply:
maybe my use of AvailableParticipantDTO is confusing. My understanding is all we need to do once we have the SiteDTO is to retrieve the participant name & logo by joining to our Participants on SiteId. If there is no associated siteId in our db, then we just display the name of the site (I suppose). I was thinking this could reuse the AvailableParticipantDTO, essentially what we need to end up with is:
- Participant Name
- Types
- Logo
- Id (not strictly necessary but I think makes sense).

fwiw by the time we get to BulkAddPermissions I don't think we even need SiteId, since this control is type-driven rather than siteId-driven.

How's that sound?

Reply:
Oh I see, thanks for the explanation, re-use AvailableParticipantDTO make sense in this case